### ISSUE TITLE
Very slow interaction with `->getDoctrineColumn` after 3.4

### ISSUE THREAD

**Issue by pxpm** [doctrine/dbal#5658]
Hello guys, thanks for this project 👍 

MYSQL 5.7
PHP 8.1

- Installed dbal v3.4 and my page went from 6s loading time, to 20s loading time. 
- Reverted to dbal v3.3 and my page went back to 6s. 

The function that we use that causes this slowness is `getDoctrineColumn()`. (just use a loop and call this 20/30 times)

Installing mysql 8.0 also fixes the problem when using dbal 3.4. 

Not sure what changed in internals and I have no time to dig it, just want to let you guys be aware of this issue.

If I can help with something else let me know.

Cheers

**Comment by pxpm** [comment]
This is probably related with https://github.com/doctrine/dbal/issues/5638

So I am going to close this.

Sorry I missed that issue at first 🙃 

Cheers

**Comment by pxpm** [comment]
Sorry, I re-opened because the issue I mentioned is already closed and I can still experience this in 3.4.4

**Comment by morozov** [comment]
@pxpm what does `getDoctrineColumn()` do and where is it defined? Alternatively, could you identify the query that takes longer than expected?

I see that the `selectTableColumns()` method of the MySQL schema manager uses a proper implementation as of https://github.com/doctrine/dbal/pull/5639.

The only remaining query that uses the schema name in a JOIN condition is the one used by `selectForeignKeyColumns()`: https://github.com/doctrine/dbal/blob/60a98765bbced7e58c0501f5156aaf2c821ec71e/src/Schema/MySQLSchemaManager.php#L482-L485

**Comment by pxpm** [comment]
@morozov I am sorry, I should have mentioned that I am using Laravel, so probably something not related with dbal directly. 
Still unknown to me why changing from dbal 3.3 to 3.4 raised the issue, but I decided to not follow the debug route and instead I reimplemented my solution using dbal classes directly. Everything working like a charm 🙏 

Thanks for the time spent on this issue 🙏 

Will be closing this, cheers!
